sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -31,7 +31,12 @@ namespace Azure { namespace Storage { namespace _internal {
   FileReader::FileReader(const std::string& filename)
   {
     int sizeNeeded = MultiByteToWideChar(
-        CP_UTF8, MB_ERR_INVALID_CHARS, filename.data(), int(filename.length()), nullptr, 0);
+        CP_UTF8,
+        MB_ERR_INVALID_CHARS,
+        filename.data(),
+        static_cast<int>(filename.length()),
```

Comment:
Why is this cast safe to do? Isn't the filename provided by the end user? There should be a static assert here or somewhere up the call stack ensuring the filename is small enough.

Reply:
Filename larger than `INT_MAX` doesn't make any sense on any OS or filesystem. So I think this case is safe.